src/main/java/com/example/nlearn/services/TestService.java

```diff
@@ -0,0 +1,119 @@
+package com.example.nlearn.controllers;
+
+import com.example.nlearn.models.FullTest;
+import com.example.nlearn.models.Question;
+import com.example.nlearn.models.Test;
+
+import com.example.nlearn.repos.GroupRepository;
+import com.example.nlearn.repos.QuestionRepository;
+import com.example.nlearn.repos.TestRepository;
+import com.example.nlearn.repos.UserRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.DeleteMapping;
+
+import java.util.List;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/tests")
+public class TestController {
```

Comment:
Rename to TestService and delete all `RestController`-related annotations
Leave only pure logic method